src/main/java/Duke.java

```diff
@@ -1,10 +1,129 @@
+                + "|  | |  || ||  <    | |\n"
+                + "|__/ |__||_||_| |_| |_|\n";
+        System.out.println("Hello from\n" + logo +"\n" + BORDER_DECORATION);
+        System.out.println("Hello! I'm Niki\nWhat can I do for you?\n" + BORDER_DECORATION);
+    }
+
+    public static void printList() {
+        for (int i = 0; i < Task.getNumberOfTasks(); i++) {
+            System.out.println((i + 1) + "."+ taskLists[i].toString());
+        }
+    }
+
+    public static void addTask(String userInput) {
+        String[] arrayOfTaskStrings = userInput.split(" ");
+        String extractTaskDescription = "";
+        for (int i = 1; i < arrayOfTaskStrings.length; i++) {
+            extractTaskDescription += arrayOfTaskStrings[i] + " ";
+        }
+
+        Task newTask = new Todo(extractTaskDescription);
+        taskLists[Task.getNumberOfTasks() - 1] = newTask;
+        System.out.println("Task added:\n\t" + newTask.toString());
+        System.out.println("Now you have " + Task.getNumberOfTasks() + " tasks in your list!");
+    }
+
+    public static void addTaskWithTime(String userInput, String stringSeparator) {
+        String[] arrayOfTaskStrings = userInput.split(" ");
+        String extractStringsWithoutCommand = "";
+        for (int i = 1; i < arrayOfTaskStrings.length; i++) {
+            extractStringsWithoutCommand += arrayOfTaskStrings[i] + " ";
+        }
+
+        String extractTaskDescription = extractStringsWithoutCommand.split(stringSeparator)[0];
+        String extractTaskDeadlineTime = extractStringsWithoutCommand.split(stringSeparator)[1];
+        Task newTask;
+        if (stringSeparator.equals("/by ")) {
+            newTask = new Deadline(extractTaskDescription, extractTaskDeadlineTime);
+        } else {
+            newTask = new Event(extractTaskDescription, extractTaskDeadlineTime);
+        }
+
+        taskLists[Task.getNumberOfTasks() - 1] = newTask;
+        System.out.println("Task added:\n\t" + newTask.toString());
+        System.out.println("Now you have " + Task.getNumberOfTasks() + " tasks in your list!");
+    }
+
+    public static boolean isWithinTaskRange(int taskNumber) {
+        if (taskNumber > Task.getNumberOfTasks() || taskNumber <= 0) {
+            System.out.println("Task does not exist!");
+            return false;
+        }
+        return true;
+    }
+
+    public static void markTask(boolean isMarked, String userInput) {
+        int taskNumber = Integer.parseInt(userInput.split(" ")[1]);
+        if (isWithinTaskRange(taskNumber)) {
+            if (isMarked) {
+                taskLists[taskNumber - 1].markAsDone();
+                System.out.println("Fantastic! This task is done:");
+            } else {
+                taskLists[taskNumber - 1].markAsUndone();
+                System.out.println("Uh oh! This task is undone:");
+            }
+            System.out.println(taskLists[taskNumber - 1].toString());
+        }
+    }
```

Comment:
Perhaps you can consider moving task management-related logic code into a class of its own (e.g. `addTask`, `isWithinTaskRange`, and `markTask` can be moved into a new class whose purpose is solely to manage tasks). 

This way, your program structure will look better. 🙂